x-pack/plugin/apm-data/src/main/resources/ingest-pipelines/apm@pipeline.yaml

```diff
@@ -24,3 +24,13 @@ processors:
     target_field: user_agent
     ignore_failure: true
     ignore_missing: true
+- script:
+    description: Add time when event was ingested (and remove sub-seconds to improve storage efficiency)
+    tag: truncate-subseconds-event-ingested
+    ignore_failure: true
+    source: |-
+      if (ctx?.event == null) {
+        ctx.event = [:];
+      }
+
+      ctx.event.ingested = metadata().now.withNano(0).format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
```

Comment:
Is there a way to _not_ set the field if it's already present?

Reply:
Yes I would like to preserve it if it's already set

Reply:
I tested 2 solutions for this:
```painless
if (!ctx.event.containsKey('ingested')) {
  ctx.event.ingested = metadata().now.withNano(0).format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
}
```

and
```painless
if (ctx.event?.ingested == null) {
  ctx.event.ingested = metadata().now.withNano(0).format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
}
```

Both worked when added to the script in preventing the pipeline to override the field.

Reply:
Nice thanks!
I'll incorporate those, as I think it's reasonable to preserve the field when already set. 🙏🏼

Reply:
The Fleet ingest does not do that, and I don't think we should be diverging. Is there a reason we can't or shouldn't use the exact same script processor as Fleet?

https://github.com/elastic/kibana/blob/e103a253d9b756605dbeb92955ff517597055970/x-pack/plugins/fleet/server/constants/fleet_es_assets.ts#L140-L149